bin/migrate-oats-data/noi/set_notice_of_intent_visibility.py

```diff
@@ -0,0 +1,160 @@
+                updated_visible_to_portal_count + updated_hidden_from_portal_count
+            )
+
+            logger.info(
+                f"Total hide_from_portal flag on notice_of_intent updated: {updated_total_count}"
+            )
+        conn.commit()
+    except Exception as err:
+        logger.exception(err)
+        cursor.close()
+        conn.close()
+
+    logger.info(f"Finished {etl_name}")
+
+
+def _set_source_on_notice_of_intent(conn):
+    """"""
+    logger.info(f"Start {etl_name} => _set_source_on_notice_of_intent")
+
+    try:
+        with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            count_created_by_applicant_query = f"""    
+                                SELECT count(*) FROM oats.oats_alr_applications oaa 
+                                JOIN alcs.notice_of_intent noi ON noi.file_number = oaa.alr_application_id::TEXT
+                                WHERE oaa.who_created = 'PROXY_OATS_APPLICANT';
+                            """
+            cursor.execute(count_created_by_applicant_query)
+            created_by_applicant_count = dict(cursor.fetchone())["count"]
+
+            count_created_by_oats_query = f"""    
+                                SELECT count(*) FROM oats.oats_alr_applications oaa 
+                                JOIN alcs.notice_of_intent noi ON noi.file_number = oaa.alr_application_id::TEXT
+                                WHERE oaa.who_created <> 'PROXY_OATS_APPLICANT';
+                            """
+            cursor.execute(count_created_by_oats_query)
+            created_by_oats_count = dict(cursor.fetchone())["count"]
+
+            count_created_in_alcs_query = f"""    
+                                SELECT count(*) FROM alcs.notice_of_intent a
+                                WHERE a."source" <> 'APPLICANT' 
+                                AND a.audit_created_by <> 'oats_etl';
+                            """
+            cursor.execute(count_created_in_alcs_query)
+            created_by_alcs_count = dict(cursor.fetchone())["count"]
+
+            count_total = (
+                created_by_applicant_count
+                + created_by_oats_count
+                + created_by_alcs_count
+            )
+
+            logger.info(f"Total application to set hide_from_portal: {count_total}")
+
+            update_created_by_applicant_query = f"""
+                                UPDATE alcs.notice_of_intent 
+                                SET "source" = 'APPLICANT'
+                                FROM oats.oats_alr_applications oaa 
+                                WHERE alcs.notice_of_intent.file_number = oaa.alr_application_id::TEXT 
+                                AND oaa.who_created = 'PROXY_OATS_APPLICANT';
+                            """
+            cursor.execute(update_created_by_applicant_query)
+            updated_created_by_applicant_count = cursor.rowcount
+
+            update_created_by_oats_query = f"""
```

Comment:
This query and the one above almost look identical, it seems they are updating the same IDs with slightly different source options. How does this work?

Reply:
I believe the key here is who_created = PROXY vs <> PROXY

Reply:
exactly the one above selects all records created by proxy, where this one everything that is NOT selected by proxy. PROXY_OATS_APPLICANT in oats means that the record was created by user from Oats Portal